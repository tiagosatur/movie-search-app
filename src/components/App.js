import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { PATHS } from '../utils/variables';
import { MainNav, PrivateRoute, Grid } from '.'
import { Home, Movie, Dashboard, Login } from '../pages';
import { GlobalStyle } from '../style/base/global';

const App = () => {
    return(
        <div className='app'>
            <GlobalStyle />
            <MainNav />
            <Grid>
                <Switch>
                    <Route exact path={PATHS.HOME.PATH} component={Home} />
                    <Route path={PATHS.MOVIE.PATH} component={Movie} />
                    <PrivateRoute component={Dashboard} /> 
                    <Route path={PATHS.LOGIN.PATH} component={Login} />
                </Switch>
            </Grid>
        </div>
    )
}


export default App;