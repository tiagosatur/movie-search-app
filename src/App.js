import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { PATHS } from './utils/variables';
import { MainNav, PrivateRoute, Grid } from './components'
import { Home, Movie, Dashboard, Login, NotFound } from './pages';
import { GlobalStyle } from './style/base/global';

const App = () => {
    return(
        <div className='app'>
            <GlobalStyle />
            <MainNav />
            <Grid>
                <Switch>
                    <Route exact path={PATHS.HOME.PATH} component={Home} />
                    <PrivateRoute path={PATHS.DASHBOARD.PATH} component={Dashboard} /> 
                    <Route path={PATHS.MOVIE.PATH} component={Movie} />
                    <Route path={PATHS.LOGIN.PATH} component={Login} />
                    <Route component={NotFound} />
                </Switch>
            </Grid>
        </div>
    )
}


export default App;