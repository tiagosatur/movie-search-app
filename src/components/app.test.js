import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';

describe('App tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <App />
    );
  });
  
  it('should render the App element', () => {
    expect(wrapper.find('.app').exists()).toBe(true);
  });

  it('should call App snapshot correctly', () => {
    const tree = renderer
        .create(wrapper)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });
})