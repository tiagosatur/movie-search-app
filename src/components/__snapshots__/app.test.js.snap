// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should call App snapshot correctly 1`] = `null`;
